### ISSUE TITLE
[lines-between-class-members] separate config for class member variables

### ISSUE THREAD

**Issue by kkimdev** [eslint/eslint#11745]
**What rule do you want to change?**
Currently, [lines-between-class-members](https://eslint.org/docs/rules/lines-between-class-members) applies uniformly to class functions and class member variables.  It will be great if we can have a separate config for each case.

**Does this change cause the rule to produce more or fewer warnings?**
No.

**How will the change be implemented? (New option, new default behavior, etc.)?**
TBD

**Please provide some example code that this change will affect:**
**What does the rule currently do for this code?**
**What will the rule do after it's changed?**

Before, lines-between-class-members is applied to member variables and functions uniformly.
```javascript
class MaterialManager {
    memberVariable1 = "hello";

    memberVariable2 = 42;

    memberVariable3;

    function1() {
    }

    function2() {
    }
}
```

After, we can configure differently for member variables:
```javascript
class MaterialManager {
    memberVariable1 = "hello";
    memberVariable2 = 42;
    memberVariable3;

    function1() {
    }

    function2() {
    }
}
```

**Are you willing to submit a pull request to implement this change?**
No

**Comment by kaicataldo** [comment]
I understand the desire for an option to allow this, and do think it could be a good addition (pending a concrete proposal) once class field properties reach stage 4 (currently stage 3).

For the sake of maintainability, ESLint doesn’t support experimental syntax that is stage 3 and below.
